pkg/kubectl/cmd/annotate.go

```diff
@@ -193,7 +204,11 @@ func (o AnnotateOptions) RunAnnotate() error {
 			return err
 		}
 
-		name, namespace, obj := info.Name, info.Namespace, info.Object
+		obj, err := resource.AsVersionedObject([]*resource.Info{info}, false, o.outputVersion.String())
```

Comment:
Right, my bad, I was also looking at conver in kubectl. Will fix that later today.